Contest_API.md

```diff
@@ -751,8 +751,20 @@ Returned data:
 
 ### Groups
 
-Grouping of teams. At the World Finals these are the super regions, at
-regionals these are often different sites.
+Grouping of teams. At the World Finals these are the super regions; at other contests these
+may be the different sites, divisions, or types of contestants.
+
+Teams may belong to multiple groups. For instance, there may be a group for each site, a group for
+university teams, a group for corporate teams, and a group for ICPC-eligible teams. Teams could
+belong to two or three of these.
+When there are different kinds of groups for different purposes (e.g. sites vs divisions), each
+group or set of groups should have a different type attribute
+(e.g. `"type":"site"` and `"type":"division"`).
+
+Groups must exist for any combination of teams that should be ranked on a
+[group scoreboard](#group-scoreboard), which means groups may be created for combinations.
+For instance, if there's a requirement to show a scoreboard for teams in each of `D`
+divisions at every one of `S` sites, then there will also be `D`x`S` groups with a type like `"type":"site-division"`.
```

Comment:
I would suggest explicitly recommending the naming scheme `<group1>-<group2>` (i.e. "group names separated by dash") for product groups, rather than "just" recommending by example.

Reply:
Rewrote this paragraph a little, including fixing this in new commit.